addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5020,29 +5097,41 @@ registry.BackgroundToggler = SnippetOptionWidget.extend({
         return shapeOption._toggleShape();
     },
     /**
-     * Toggles background filter on or off.
+     * Removes remaining transparent bg filter
      *
-     * @see this.selectClass for parameters
+     * @override
      */
-    toggleBgFilter(previewMode, widgetValue, params) {
-        if (widgetValue) {
-            const bgFilterEl = document.createElement('div');
-            bgFilterEl.classList.add('o_we_bg_filter', 'bg-black-50');
-            const lastBackgroundEl = this._getLastPreFilterLayerElement();
-            if (lastBackgroundEl) {
-                $(lastBackgroundEl).after(bgFilterEl);
-            } else {
-                this.$target.prepend(bgFilterEl);
+    async cleanForSave() {
```

Comment:
Should not be the job of `cleanForSave` -> Your option sets a transparent div, it should remove that div by itself, not waiting for someone else to clean its mess later :wink: 

cleanForSave cases are more and more uncommon the more we improve the way the edit mode works. (Probably half of the current cases should not even use it anymore either). The only case that comes to mind is an option add *an edit mode only* class on an element to add a visual effect on an element for the time of edition -> it should be removed on save (we have that case).